### PR TITLE
[MRG] Add numba solver for cd

### PR DESCRIPTION
**Pull request by mathurinm**
Locally it's not converging on Boston for now

### CODE REVIEW

**Comment by mathurinm** [comment]
Curves look nice @tomMoral , can you review ?

File: benchmarks/lasso/solvers/cd.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class Solver(BaseSolver):
+    name = "cd"
+
+    install_cmd = None
+
+    def set_objective(self, X, y, lmbd):
+        self.X, self.y, self.lmbd = X, y, lmbd
+        # self.run(1)
```
Comment (mathurinm) [inline benchmarks/lasso/solvers/cd.py]: @tomMoral should we run 1 iteration to compile numba? Or is it neglectable ?

Comment (agramfort) [reply benchmarks/lasso/solvers/cd.py]: yes do this to cache the compiled code

File: setup.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
     url=URL,
     packages=find_packages(),
     install_requires=['numpy', 'pandas', 'matplotlib', 'pyyaml',
-                      'click', 'joblib', 'scikit-learn'],
+                      'click', 'joblib', 'scikit-learn', 'numba'],
```
Comment (agramfort) [inline setup.py]: we should avoid making these requirements global. This is solver specific

**Pull request merged**